### PR TITLE
Pushbullet channel

### PR DESCRIPTION
**Pull request by chimpchimp**
Proposed changes in this pull request:
- Add support to notify PushBullet Channels instead of just devices
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/SickRage/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked
- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by miigotu** [comment]
First and foremost, this needs rebased on develop. No-ones commits except your own should show up.

**Comment by OmgImAlexis** [comment]
The vender file should not have changed at all.

**Comment by chimpchimp** [comment]
Reverted the vendor file (not sure how that got changed) and rebased.

**Comment by chimpchimp** [comment]
Just noticed there was another merge in there, I'll fix.

**Comment by chimpchimp** [comment]
There we go. Now everything is clean.

**Comment by OmgImAlexis** [comment]
You may want to also rename those commits as they don't really explain what you did. Also I'd suggest just squashing them into a single commit.

File: gui/slick/js/core.js [hunk 1]
```diff
@@ -936,6 +936,36 @@ var SICKRAGE = {
                     $("#pushbullet_device").val($("#pushbullet_device_list").val());
                     $('#testPushbullet-result').html("Don't forget to save your new pushbullet settings.");
                 });
+
+                $.get(srRoot + "/home/getPushbulletChannels", {
+                    'api': pushbullet.api
+                }, function (data) {
+                    pushbullet.channels = $.parseJSON(data).channels;
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js:943]: Not sure if the currently js we use does this or not but can you not just use getJSON and skip this?

File: gui/slick/js/core.js [hunk 2]
```diff
@@ -936,6 +936,36 @@ var SICKRAGE = {
...
+                        $("#pushbullet_channel_list").prop('disabled', true);
+                    }
+                    if (msg) {
+                        $('#testPushbullet-result').html(msg);
+                    }
+
+                    $("#pushbullet_channel_list").on('change', function () {
+                        $("#pushbullet_channel").val($("#pushbullet_channel_list").val());
+                        $('#testPushbullet-result').html("Don't forget to save your new pushbullet settings.");
+                });
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js]: This should be indented to be inline correctly.

File: gui/slick/js/core.js [hunk 3]
```diff
@@ -2602,8 +2642,16 @@ var SICKRAGE = {
...
+<<<<<<< HEAD
                 $.post(srRoot + '/manage/failedDownloads', 'toRemove='+removeArr.join('|'));
+<<<<<<< HEAD
                 location.reload(true);
+=======
+                location.reload();
+>>>>>>> e5b1aad... revert dumb formatting
+=======
+                $.post(srRoot + '/manage/failedDownloads', 'toRemove='+removeArr.join('|'), function() { location.reload(true); });
+>>>>>>> ebf9ac9... revert accidental changes
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js]: All of the merges need to be fixed.

**Comment by miigotu** [comment]
I have to look it over also =p

File: gui/slick/js/core.js (hunk 1 above)
Comment (chimpchimp) [inline gui/slick/js/core.js:943]: The current JS does when checking for devices so I pretty much just copied it for consistency. I could got back and clean up both the original step and my own.

Comment (OmgImAlexis) [inline gui/slick/js/core.js:943]: All's good I'll change it over later when I can up the `core.js` file.

File: gui/slick/js/core.js (hunk 3 above)
Comment (chimpchimp) [inline gui/slick/js/core.js]: Cleaned this up on a new commit and re-minified to be safe.

File: gui/slick/js/core.js [hunk 4]
```diff
@@ -2604,6 +2637,8 @@ var SICKRAGE = {
 
                 $.post(srRoot + '/manage/failedDownloads', 'toRemove='+removeArr.join('|'));
                 location.reload(true);
+                location.reload();
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js]: why is there two location reloads here?

File: gui/slick/js/core.js [hunk 5]
```diff
@@ -2604,6 +2637,8 @@ var SICKRAGE = {
...
+                $.post(srRoot + '/manage/failedDownloads', 'toRemove='+removeArr.join('|'), function() { location.reload(true); });
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js]: Same as this?

File: gui/slick/js/core.js [hunk 6]
```diff
@@ -2158,9 +2188,12 @@ var SICKRAGE = {
                 });
 
                 if (epArr.length === 0) { return false; }
-
                 $.post(srRoot + '/home/setStatus', 'show=' + $('#showID').attr('value') + '&eps=' + epArr.join('|') + '&status=' + $('#statusSelect').val());
                 location.reload(true);
+                location.reload();
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js]: and here.

File: gui/slick/js/core.js (hunk 4 above)
Comment (chimpchimp) [inline gui/slick/js/core.js]: I actually don't know how that got added since that section doesn't have anything to do with what I added.

**Comment by chimpchimp** [comment]
Cleaned up all the weird JS stuff that I assume was from my initial failed merges/rebases.

**Comment by miigotu** [comment]
"This branch has conflicts that must be resolved"

**Comment by OmgImAlexis** [comment]
@chimpchimp you might also want to look at your git settings since `ryan@ryanzisk.com` isn't associated with your Github account but is used on all of these commits.

**Comment by miigotu** [comment]
If you dont have the upstream remote configured you should

``` bash
git remote add upstream https://github.com/SickRage/SickRage.git
```

then:

``` bash
git fetch upstream
git rebase -i upstream/develop
(fix conflicts)
git commit -a
git push origin pushbullet-channel -f
```

**Comment by chimpchimp** [comment]
Conflicts should be resolved now, had to run the grunt jobs again.

File: gui/slick/js/core.js [hunk 7]
```diff
@@ -936,6 +936,36 @@ var SICKRAGE = {
...
+                    $("#pushbullet_channel_list").html('');
+                    if (pushbullet.channels.length > 0) {
+                        for (var i = 0, len = pushbullet.channels.length; i < len; i++) {
+                            if (pushbullet.channels[i].active === true) {
+                                $("#pushbullet_channel_list").append('<option value="' + pushbullet.channels[i].tag + '" selected>' + pushbullet.channels[i].name + '</option>');
+                            } else {
+                                $("#pushbullet_channel_list").append('<option value="' + pushbullet.channels[i].tag + '">' + pushbullet.channels[i].name + '</option>');
+                            }
+                        }
+                        $("#pushbullet_channel_list").prepend('<option value="" ' + (pushbullet.currentChannel === '' ? 'selected' : '') + '>No Channel</option>');
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js]: This can be changed to `pushbullet.currentChannel ? 'selected' : ''` since an empty string is false.

Comment (chimpchimp) [inline gui/slick/js/core.js]: Corrected in newest push.

**Comment by OmgImAlexis** [comment]
That looks fine to me, @miigotu QuantifiedCode has an issue and pullapprove and coveralls are frozen.

**Comment by miigotu** [comment]
@chimpchimp can you add a docstring to `get_channels`(and any other new methods)? I know we are missing docstrings all over the place but we are trying to work on that =P. It's the 1 issue quantifiedcode is reporting.

About coveralls, there have been some issues with travis calling coveralls -.-

**Comment by chimpchimp** [comment]
Just pushed up a new commit and it looks like the Travis job failed this time but if I'm reading the log correctly it is possibly related to pushing to coveralls?

**Comment by miigotu** [comment]
Yeah travis is having some issues with our tox setup and coveralls....

**Pull request merged**